pkg/appgw/frontend_listeners.go

```diff
@@ -23,16 +23,19 @@ func (c *appGwConfigBuilder) getListeners(cbCtx *ConfigBuilderContext) (*[]n.App
 	}
 
 	publIPPorts := make(map[string]string)
-	portSet := make(map[string]interface{})
+	portsByNumber := cbCtx.ExistingPortsByNumber
 	var listeners []n.ApplicationGatewayHTTPListener
-	var ports []n.ApplicationGatewayFrontendPort
+
+	if portsByNumber == nil {
+		portsByNumber = make(map[Port]n.ApplicationGatewayFrontendPort)
+	}
 
 	if cbCtx.EnvVariables.EnableIstioIntegration {
-		listeners, ports, portSet, publIPPorts = c.getIstioListenersPorts(cbCtx)
+		listeners, portsByNumber, publIPPorts = c.getIstioListenersPorts(cbCtx)
```

Comment:
will this overwrite portsByNumber assigned at line 26 ?

Reply:
Thanks for catching that!